conjure-core/src/main/java/com/palantir/conjure/defs/validator/EndpointDefinitionValidator.java

```diff
@@ -367,30 +367,33 @@ public void validate(EndpointDefinition definition) {
     @com.google.errorprone.annotations.Immutable
     private static final class ParamIdValidator implements ConjureValidator<EndpointDefinition> {
         @Override
+        @SuppressWarnings("Slf4jLogsafeArgs")
         public void validate(EndpointDefinition definition) {
             definition.getArgs().forEach(arg -> {
-                final Pattern pattern;
                 ParameterType paramType = arg.getParamType();
-                if (paramType.accept(ParameterTypeVisitor.IS_BODY)
-                        || paramType.accept(ParameterTypeVisitor.IS_PATH)
-                        || paramType.accept(ParameterTypeVisitor.IS_QUERY)) {
-                    pattern = ANCHORED_PATTERN;
+                if (paramType.accept(ParameterTypeVisitor.IS_BODY) || paramType.accept(ParameterTypeVisitor.IS_PATH)) {
+                    // No validation
```

Comment:
We should still validate body and path params

Reply:
Previously we would set the `pattern` variable for path/body params but never use it (the if statements below only matched header and query). I assume this is because path and body params don’t have a serialized key, so the param ID is irrelevant

Reply:
I can make this clearer in the comment